### PR TITLE
Recurrence Operators and Recurrence Relations for Holonomic Functions.

### PR DESCRIPTION
**Pull request by shubhamtibra**
- [x] Added Recurrence Operators
- [x] Find Recurrence Relation in Power series coefficients of Holonomic Functions.
- [x] Power series expansion.
- [ ] Numerical Computation of Holonomic Functions.
- [x] Tests

### CODE REVIEW

**Comment by shubhamtibra** [comment]
We first need to merge [this](https://github.com/sympy/sympy/pull/11115) PR for robustness. This is an algebraic representation of Recurrence Operators pretty much similar to Differential Operators.

**Comment by shubhamtibra** [comment]
Ping @jksuom @certik .

**Comment by shubhamtibra** [comment]
I have added the method to find Recurrence relation between coefficients of Taylor expansion of Holonomic function. Will soon add Initial conditions support for the Recurrence relation.

File: sympy/holonomic/holonomic.py [hunk 1]
```diff
@@ -786,6 +787,31 @@ def composition(self, expr):
             return HolonomicFunction(sol, self.x, self.x0, self.y0)
         return HolonomicFunction(sol, self.x)
 
+    def to_Sequence(self):
```
Comment (jksuom) [inline sympy/holonomic/holonomic.py]: I think this should be `to_sequence` according to [SymPy conventions](https://github.com/sympy/sympy/wiki/Development-workflow#id23).

File: sympy/holonomic/recurrence.py [hunk 2]
```diff
@@ -319,3 +319,36 @@ def __eq__(self, other):
...
+            self.u0 = u0
+
+        if len(self.u0) == 0:
+            self._have_init_cond = False
+        else:
+            self._have_init_cond = True
+        self.n = recurrence.parent.base.gens[0]
+
+    def __repr__(self):
+
```
Comment (jksuom) [inline sympy/holonomic/recurrence.py]: No empty line here

Comment (shubhamtibra) [inline sympy/holonomic/recurrence.py]: Does this applies to all functions? I had added empty lines after some other functions also.

Comment (jksuom) [inline sympy/holonomic/recurrence.py]: I believe so, at least if there is no docstring. PEP8 only says this:
`Use blank lines in functions, sparingly, to indicate logical sections.`

Comment (shubhamtibra) [inline sympy/holonomic/recurrence.py]: I'll take care of it.

File: sympy/holonomic/holonomic.py [hunk 3]
```diff
@@ -786,6 +788,50 @@ def composition(self, expr):
...
+                if coeff == 0:
+                    continue
+                if i - k in dict1.keys():
+                    dict1[i - k] += (coeff * factorial(n - k + i) / factorial(n - k)).simplify()
+                else:
+                    dict1[i - k] = (coeff * factorial(n - k + i) / factorial(n - k)).simplify()
+        sol = []
+        lower = min(dict1.keys())
+        upper = max(dict1.keys())
+        for i in range(upper - lower + 1):
```
Comment (jksuom) [inline sympy/holonomic/holonomic.py]: You could probably have `for j in range(lower, upper + 1):`

**Pull request merged**